src/google/cloud/ndb/model.py

```diff
@@ -642,7 +642,8 @@ def _entity_to_ds_entity(entity, set_key=True):
     if set_key:
         key = entity._key
         if key is None:
-            key = key_module.Key(entity._get_kind(), None)
+            # use _class_name instead of _get_kind, to get PolyModel right
```

Comment:
Why? Is it possible that _get_kind is misbehaving and/or needs to be overridden for PolyModel?

Reply:
_class_name is just an alias for _get_kind, except in PolyModel, where _get_kind returns the base PolyModel class and _class_name returns the actual class name. This is by design.

For this test, using _get_kind here would store the PolyModel entity as class Animal, instead of the correct value, which is Cat. Calling _class_name returns the correct value for PolyModel, while using it for Model or Expando will be the same as calling _get_kind for them.

Hopefully I'm not confusing you more.

Reply:
One thing more. It's not enough to override _get_kind for PolyModel, because the API for PolyModel requires that calling kind() for all classes in the hierarchy returns the same base class. Hence the need for _class_name.

Reply:
Ok, thanks for explaining that.